pkg/telemetry/phonehome/segment/segment.go

```diff
@@ -0,0 +1,137 @@
+package segment
+
+import (
+	segment "github.com/segmentio/analytics-go"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// Enabled tells whether telemetry data collection is enabled.
+func Enabled() bool {
+	return env.TelemetryStorageKey.Setting() != ""
+}
+
+type segmentTelemeter struct {
+	client   segment.Client
+	userID   string
+	identity map[string]any
+}
+
+// NewTelemeter creates and initializes a Segment telemeter instance.
+func NewTelemeter(userID string, identity map[string]any) *segmentTelemeter {
+	key := env.TelemetryStorageKey.Setting()
+	server := env.TelemetryEndpoint.Setting()
+	return initSegment(userID, identity, key, server)
+}
+
+type logWrapper struct {
+	internal *logging.Logger
+}
+
+func (l *logWrapper) Logf(format string, args ...any) {
+	l.internal.Logf(zapcore.InfoLevel, format, args...)
+}
+
+func (l *logWrapper) Errorf(format string, args ...any) {
+	l.internal.Errorf(format, args...)
+}
+
+func initSegment(userID string, identity map[string]any, key, server string) *segmentTelemeter {
+	segmentConfig := segment.Config{
+		Endpoint: server,
+		Interval: env.TelemetryFrequency.DurationSetting(),
+		Logger:   &logWrapper{internal: log},
```

Comment:
We use the `proxy.RoundTripper` to ensure that the transport picks up all relevant proxy settings (including the `all_proxy` setting):
```suggestion
		Interval: env.TelemetryFrequency.DurationSetting(),
		Transport: proxy.RoundTripper(),
		Logger:   &logWrapper{internal: log},
```